parsers/US_SPP.py

```diff
@@ -341,12 +341,12 @@ def fetch_live_exchange(
 
 
 def fetch_historical_exchange(
-    zone_key1: str,
-    zone_key2: str,
+    zone_key1: ZoneKey,
+    zone_key2: ZoneKey,
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> list:
+) -> list[dict[str, Any]]:
     filename = target_datetime.strftime("TieFlows_%b%Y.csv")
     file_url = f"{US_PROXY}/file-browser-api/download/historical-tie-flow?{HOST_PARAMETER}&path={filename}"
```

Comment:
Not really part of your changes but I donät think the target datetime should be None here, type wise. Or it should be checked that it's not none before the filename is generated as `.strftime` is not available on `None`.

Reply:
Fixed both! Sorry it's been a while!